src/v/cluster/topic_recovery_validator.cc

```diff
@@ -0,0 +1,376 @@
+#include "topic_recovery_validator.h"
+
+#include "cloud_storage/anomalies_detector.h"
+#include "cluster/logger.h"
+
+#include <seastar/coroutine/as_future.hh>
+
+// TODO design alternative: it's possible to use this subclass of
+// abort_requested_exception to support early exit in validation. `struct
+// terminate_early_exception final : public ss::abort_requested_exception {};`
+// do_validate_recovery_partition could terminate siblings operating on
+// different partisions like this:
+// `as.request_abort_ex(terminate_early_exception{})`
+// and while handling remote request, terminate_early_exceptions should be
+// flattened to a generic "eary_termination"  like so `co_await
+// remote.manifest_exists(bucket, path,
+// rtc).handle_exception_type([](terminate_early_exception) { return
+// early_termination; });` on the caller side, this local abort_source will
+// propagate root abort requests and local terminate_early requests
+// `ss::abort_source validation_as{}; auto master_as_subscription =
+// as.subscribe([&] { validation_as.request_abort(); });` ultimately this design
+// is not implemented here, since there seems no concrete use case to terminate
+// early at this level. either it's a general shutdown case, or the caller code
+// can get a better sense of what's going on by knowning the result of all
+// partitions
+
+namespace cluster {
+
+auto do_validate_manifest_existence(
+  cloud_storage_clients::bucket_name const& bucket,
+  model::ntp const& ntp,
+  model::initial_revision_id rev_id,
+  cloud_storage::remote& remote,
+  retry_chain_node& rtc) -> ss::future<validation_result> {
+    // check manifest existence, handle download_result::success as Found
+    // and the rest as NotFound (but warn for failed and timedout).
+    // This existence check could be flattened in anomalies_detector, but
+    // it's not a perfect overlap with the functionalities, so it's
+    // performed externally
+    using enum validation_result;
+
+    return remote.partition_manifest_exists(bucket, ntp, rev_id, rtc)
+      .then(
+        [ntp, rev_id](cloud_storage::remote::partition_manifest_existence res) {
+            auto [download_res, _] = res;
+
+            // only res==success (manifest found) or res==notfound (manifest NOT
+            // found) make sense. warn for timedout and failed
+            if (download_res == cloud_storage::download_result::success) {
+                vlog(clusterlog.info, "{}/{} manifest found", ntp, rev_id);
```

Comment:
my reasoning is that during recovery I want to see a trace of the recovery, 
so that we get

INFO ... Performing validation {} on topic {} with {} partitions
and for each partition either an INFO line or an ERROR line

Reply:
This makes sense. I think that the message should say something more than "manifest found", though. Put it in context--in reality there might be other stuff intermingled in the log, and "manifest found" isn't very self-describing.